### ISSUE TITLE
VyperVPN download recipe no longer works

### ISSUE THREAD

**Issue by flammable** [autopkg/homebysix-recipes#660]
Howdy! It looks like developer has changed the URL pattern for VyperVPN. Here's the page with the direct link:

https://www.vyprvpn.com/vpn-apps/vpn-for-mac

Super verbose output:

```
% autopkg run -vvv /Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes/VyprVPN/VyprVPN.download.recipe 
Processing /Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes/VyprVPN/VyprVPN.download.recipe...
WARNING: /Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes/VyprVPN/VyprVPN.download.recipe is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
{'AUTOPKG_VERSION': '2.7.3',
 'GIT_PATH': '/Library/Developer/CommandLineTools/usr/bin/git',
 'MUNKI_REPO': '/Volumes/munki_repo',
 'NAME': 'VyprVPN',
 'PARENT_RECIPES': [],
 'RECIPE_CACHE_DIR': '/Users/flammable/Library/AutoPkg/Cache/com.github.homebysix.download.VyprVPN',
 'RECIPE_DIR': '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes/VyprVPN',
 'RECIPE_OVERRIDE_DIRS': ['~/Library/AutoPkg/RecipeOverrides'],
 'RECIPE_PATH': '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes/VyprVPN/VyprVPN.download.recipe',
 'RECIPE_REPOS': {'/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.Vinrobot-recipes': {'URL': 'https://github.com/autopkg/Vinrobot-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.aanklewicz-recipes': {'URL': 'https://github.com/autopkg/aanklewicz-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.andrewvalentine-recipes': {'URL': 'https://github.com/autopkg/andrewvalentine-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.apettinen-recipes': {'URL': 'https://github.com/autopkg/apettinen-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.apizz-recipes': {'URL': 'https://github.com/autopkg/apizz-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.arubdesu-recipes': {'URL': 'https://github.com/autopkg/arubdesu-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.aysiu-recipes': {'URL': 'https://github.com/autopkg/aysiu-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.clburlison-recipes': {'URL': 'https://github.com/autopkg/clburlison-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes': {'URL': 'https://github.com/autopkg/dataJAR-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.eholtam-recipes': {'URL': 'https://github.com/autopkg/eholtam-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.flammable-recipes': {'URL': 'https://github.com/autopkg/flammable-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.foigus-recipes': {'URL': 'https://github.com/autopkg/foigus-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.gerardkok-recipes': {'URL': 'https://github.com/autopkg/gerardkok-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes': {'URL': 'https://github.com/autopkg/grahamgilbert-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.hansen-m-recipes': {'URL': 'https://github.com/autopkg/hansen-m-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.hjuutilainen-recipes': {'URL': 'https://github.com/autopkg/hjuutilainen-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes': {'URL': 'https://github.com/autopkg/homebysix-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jaharmi-recipes': {'URL': 'https://github.com/autopkg/jaharmi-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes': {'URL': 'https://github.com/autopkg/jazzace-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jessepeterson-recipes': {'URL': 'https://github.com/autopkg/jessepeterson-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jleggat-recipes': {'URL': 'https://github.com/autopkg/jleggat-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.joshua-d-miller-recipes': {'URL': 'https://github.com/autopkg/joshua-d-miller-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jps3-recipes': {'URL': 'https://github.com/autopkg/jps3-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.justinrummel-recipes': {'URL': 'https://github.com/autopkg/justinrummel-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.keeleysam-recipes': {'URL': 'https://github.com/autopkg/keeleysam-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.kevinmcox-recipes': {'URL': 'https://github.com/autopkg/kevinmcox-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.killahquam-recipes': {'URL': 'https://github.com/autopkg/killahquam-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.luisgiraldo-recipes': {'URL': 'https://github.com/autopkg/luisgiraldo-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.macprince-recipes': {'URL': 'https://github.com/autopkg/macprince-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.munkireport-recipes': {'URL': 'https://github.com/autopkg/munkireport-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.natewalck-recipes': {'URL': 'https://github.com/autopkg/natewalck-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.nmcspadden-recipes': {'URL': 'https://github.com/autopkg/nmcspadden-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.nstrauss-recipes': {'URL': 'https://github.com/autopkg/nstrauss-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.orbsmiv-recipes': {'URL': 'https://github.com/autopkg/orbsmiv-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.patgmac-recipes': {'URL': 'https://github.com/autopkg/patgmac-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes': {'URL': 'https://github.com/autopkg/recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.rtrouton-recipes': {'URL': 'https://github.com/autopkg/rtrouton-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.scriptingosx-recipes': {'URL': 'https://github.com/autopkg/scriptingosx-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.sheagcraig-recipes': {'URL': 'https://github.com/autopkg/sheagcraig-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.tallfunnyjew-recipes': {'URL': 'https://github.com/autopkg/tallfunnyjew-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.thenikola-recipes': {'URL': 'https://github.com/autopkg/thenikola-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.timsutton-recipes': {'URL': 'https://github.com/autopkg/timsutton-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.valdore86-recipes': {'URL': 'https://github.com/autopkg/valdore86-recipes'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.wardsparadox-recipes': {'URL': 'https://github.com/autopkg/wardsparadox-recipes.git'},
                  '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.facebook.Recipes-for-AutoPkg': {'URL': 'https://github.com/facebook/Recipes-for-AutoPkg.git'}},
 'RECIPE_SEARCH_DIRS': ['.',
                        '~/Library/AutoPkg/Recipes',
                        '/Library/AutoPkg/Recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.Vinrobot-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.andrewvalentine-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.apizz-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.arubdesu-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.aysiu-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.clburlison-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.dataJAR-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.eholtam-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.flammable-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.foigus-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.gerardkok-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.hansen-m-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.hjuutilainen-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jaharmi-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jessepeterson-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jleggat-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.joshua-d-miller-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.jps3-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.justinrummel-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.keeleysam-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.kevinmcox-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.killahquam-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.luisgiraldo-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.macprince-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.munkireport-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.natewalck-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.nmcspadden-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.nstrauss-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.orbsmiv-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.patgmac-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.rtrouton-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.scriptingosx-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.sheagcraig-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.tallfunnyjew-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.thenikola-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.timsutton-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.valdore86-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.wardsparadox-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.facebook.Recipes-for-AutoPkg',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.apettinen-recipes',
                        '/Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.aanklewicz-recipes'],
 'verbose': 3}
URLTextSearcher
{'Input': {'re_pattern': 'desktop/mac/production/([\\d\\.]+)/VyprVPN',
           'result_output_var_name': 'version',
           'url': 'https://www.vyprvpn.com/vpn-apps/vpn-for-mac'}}
URLTextSearcher: Found matching text (version): 5.2.0.10702
{'Output': {'version': '5.2.0.10702'}}
URLDownloader
{'Input': {'filename': 'VyprVPN-5.2.0.10702.dmg',
           'url': 'https://www.goldenfrog.com/downloads/vyprvpn/desktop/mac/production/5.2.0.10702/VyprVPN_v5.2.0.10702.dmg'}}
URLDownloader: No value supplied for prefetch_filename, setting default value of: False
URLDownloader: No value supplied for CHECK_FILESIZE_ONLY, setting default value of: False
URLDownloader: ERROR: (56) The requested URL returned error: 404

Traceback (most recent call last):
  File "/Library/AutoPkg/autopkglib/URLGetter.py", line 172, in execute_curl
    result = subprocess.run(
  File "/Library/AutoPkg/Python3/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/curl', '--silent', '--show-error', '--no-buffer', '--dump-header', '-', '--speed-time', '30', '--location', '--url', 'https://www.goldenfrog.com/downloads/vyprvpn/desktop/mac/production/5.2.0.10702/VyprVPN_v5.2.0.10702.dmg', '--fail', '--output', '/Users/flammable/Library/AutoPkg/Cache/com.github.homebysix.download.VyprVPN/downloads/tmp5xiaudxw']' returned non-zero exit status 56.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/AutoPkg/autopkglib/__init__.py", line 840, in process
    self.env = processor.process()
  File "/Library/AutoPkg/autopkglib/__init__.py", line 626, in process
    self.main()
  File "/Library/AutoPkg/autopkglib/URLDownloader.py", line 331, in main
    raw_headers = self.download_with_curl(curl_cmd)
  File "/Library/AutoPkg/autopkglib/URLGetter.py", line 186, in download_with_curl
    proc_stdout, proc_stderr, retcode = self.execute_curl(curl_cmd, text)
  File "/Library/AutoPkg/autopkglib/URLGetter.py", line 181, in execute_curl
    raise ProcessorError(e.stderr) from e
autopkglib.ProcessorError: curl: (56) The requested URL returned error: 404

  File "/Library/AutoPkg/autopkglib/__init__.py", line 840, in process
    self.env = processor.process()
curl: (56) The requested URL returned error: 404

Failed.
Receipt written to /Users/flammable/Library/AutoPkg/Cache/com.github.homebysix.download.VyprVPN/receipts/VyprVPN.download-receipt-20241015-182528.plist

The following recipes failed:
    /Users/flammable/Library/AutoPkg/RecipeRepos/com.github.autopkg.homebysix-recipes/VyprVPN/VyprVPN.download.recipe
        Error in com.github.homebysix.download.VyprVPN: Processor: URLDownloader: Error: curl: (56) The requested URL returned error: 404

Nothing downloaded, packaged or imported.
```

The actual URL is now:

`https://www.vyprvpn.com/downloads/vyprvpn/desktop/mac/production/5.2.0.10702/VyprVPN_v5.2.0.dmg`

The first part of the URL is easy, but I'm not sure how to handle the disk image's filename.

**Comment by homebysix** [comment]
This should be fixed now (#668).

**Comment by flammable** [comment]
Thanks, Elliot! 😄
